### ISSUE TITLE
Sender-dependent transports ERROR -- Relay with Amazon SES not working

### ISSUE THREAD

**Issue by pbdco** [mailcow/mailcow-dockerized#3756]
Hi, my dockerized Mailcow is working fine to RECEIVE emails, but when I set up the Sender-dependent transport (Relay) with AWS SES, this error shows up when I run the test tool:

```
No MX records for email-smtp.sa-east-1.amazonaws.com were found in DNS, skipping and using hostname as next-hop.
Connection: opening to email-smtp.sa-east-1.amazonaws.com:25, timeout=10, options=array (
    ↪ 'ssl' =>
    ↪ array (
    ↪ 'verify_peer' => false,
    ↪ 'verify_peer_name' => false,
    ↪ 'allow_self_signed' => true,
    ↪ ),
)
    ↪ Connection failed. Error #2: stream_socket_client(): unable to connect to email-smtp.sa-east-1.amazonaws.com:25 (Operation timed out) [/web/inc/lib/vendor/phpmailer/phpmailer/src/SMTP.php line 344]
    ↪ SMTP ERROR: Failed to connect to server: Operation timed out (110)
    ↪ SMTP connect() failed. https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting
```

**Things to consider:**

- Outside Mailcow, my Amazon SES SMTP credentials work 100% -for example sending with Thunderbird- with 9.9/10 deliverability rating.
- The hostname and sender I'm using to test are checked and validated on AWS SES control panel.
- Despite the error message, I tried to send an email from SOGo anyways (using the relay config for my host), but nothing happens (no email nor error messages..)
- There are no firewalls or any third party app blocking my connections.
- Mailcow is working on a standalone server with a fresh Ubuntu 18.04
- Also tried with ports:  email-smtp.sa-east-1.amazonaws.com:25, 465 and 587  -- same output.
- My docker-compose ports are as default.

**Comment by andryyy** [comment]
Issue template...

Where is it hosted?

**Comment by pbdco** [comment]
Hosted at Linode

**Comment by pbdco** [comment]
| Question | Answer |
| --- | --- |
| My operating system | Ubuntu 18.04 |
| Is Apparmor, SELinux or similar active? | NO |
| Virtualization technlogy (KVM, VMware, Xen, etc - **LXC and OpenVZ are not supported** | KVM |
| Server/VM specifications (Memory, CPU Cores) | 4gb RAM 4 CPU |
| Docker Version (`docker version`) | Docker version 19.03.12, build 48a66213fe |
| Docker-Compose Version (`docker-compose version`) | docker-compose version 1.27.0, build 980ec85b |
| Reverse proxy (custom solution) | NO |

- Output of `git diff origin/master`, any other changes to the code? NO
- All third-party firewalls and custom iptables rules are unsupported. NO Firewalls or third parties iptables, etc.
- DNS problems? Please run `docker exec -it $(docker ps -qf name=acme-mailcow) dig +short stackoverflow.com @172.22.1.254` (set the IP accordingly, if you changed the internal mailcow network) and post the output. -- No dns problems --

Output:
151.101.193.69
151.101.1.69
151.101.129.69
151.101.65.69



**docker-compose logs : HERE --> https://pastebin.com/wz1n683W**

**Comment by pbdco** [comment]
I can notice in docker logs (postfix dockerID) this repeating output:

Sep 12 14:07:16 mail postfix/smtp[678]: connect to email-smtp.sa-east-1.amazonaws.com[18.231.10.133]:25: Connection timed out

**Comment by andryyy** [comment]
https://www.linode.com/blog/linode/a-new-policy-to-help-fight-spam/

They block port 25 outbound by default.

**Comment by pbdco** [comment]
> https://www.linode.com/blog/linode/a-new-policy-to-help-fight-spam/
> 
> They block port 25 outbound by default.

No workaround? I've tried putting the hostname ports as :465 and :587 ... same output

**Comment by MAGICCC** [comment]
Read the link @andryyy gave you? Everything is explained there, even how to ask for unblock

**Comment by Adorfer** [comment]
template not used, answer not read. support ticket. closed

**Comment by pbdco** [comment]
Hi @andryyy @MAGICCC @Adorfer 

Indeed the problem was because my provider blocks the 25, 465 and 587 ports as default. I've sent them a ticket with my request and the problem has been solved. 

Everything is working 100% now and I am very happy for having chosen such a nice piece of email client. The performance, and UI/UX are amazing.

Congratulations for your great work and keep it up with such amazing project. Thank you very much for sharing it with the world.

Regards.

**Comment by BrianHenryIE** [comment]
I had the same issue but self-hosted at home on a Raspberry Pi.
I was able to send via the AWS SES SMTP server using Mail.app on my laptop in the same network.
The solution was to **set the port to 587,** i.e. `email-smtp.us-west-2.amazonaws.com:587`, as described [on the Mailcow forums](https://community.mailcow.email/d/174-relay-mail-through-external-smtp-server/3). (maybe there's also some problem with my Docker setup I should address to get the default port working)

Additionally, after configuring the Email/Routing/Sender Depending Transports, I had to configure the domain to actually use it! Email delivered to Gmail with SPF, DKIM and DMARC pass! Thanks everyone.
